packages/@angular/cli/commands/completion.ts

```diff
@@ -117,14 +117,15 @@ const CompletionCommand = Command.extend({
         });
       }
 
-      let opts = '';
+      let opts: string[] = [];
       if (command.blueprints && command.blueprints[0]) {
-        opts += extractBlueprints(command.blueprints);
+        opts = opts.concat(extractBlueprints(command.blueprints));
       }
 
       if (command.availableOptions && command.availableOptions[0]) {
-        opts += extractOptions(command.availableOptions);
-        caseBlock = caseBlock + '    ' + com.sort().join('|') + ') opts="' + opts + '" ;;\n';
+        opts = opts.concat(extractOptions(command.availableOptions));
+        const optsStr = opts.sort().join(' ');
+        caseBlock = caseBlock + '    ' + com.sort().join('|') + ') opts="' + optsStr + '" ;;\n';
```

Comment:
I second the approach with a strung interpolation:
```typescript
   caseBlock = '${caseBlock} ${com.sort().join('|')}) opts="${optsStr}" ;;\n'
```
as long as the resulting string ends in   ...) opts="........" ;;

The quotes delimiting the `opts` variable must be produced, otherwise the shell interprets  `opts`=`option_1` `newCommand` + some more garble following it, and treats the second option ("newCommand") as if it were a command with some extra arguments.